### ISSUE TITLE
replacing BTC addresses when benchmarking!

### ISSUE THREAD

**Issue by netswalker1** [nicehash/NiceHashMiner-Archived#556]
why during the benchmark my address changed to this 34HKWdzLxWBduUfJE9JxaFhoXnfC6gmepg ?

**Comment by Scorpers** [comment]
seconded, found the same in my logs.
https://www.nicehash.com/?p=miners&addr=34HKWdzLxWBduUfJE9JxaFhoXnfC6gmePG

**Comment by S74nk0** [comment]
@netswalker1 @Scorpers that is the **DEMO** user address. It is used for convenience, most new users don't have a BTC address and since some miners need a direct connection to do a benchmark this address is used because it is always known, no need for handling and prompting the users to first enter the address to do a benchmark it reduces unneeded complexity.
Closing issue...

**Comment by Scorpers** [comment]
it also enriches nicehash while forcing users to give you the CPU&GPU time every time they update the software, im certainly not benchmarking any more on your software until this is changed, one pop-up to remind the user to enter an address or to simply ASK his permission to mine to your account would have sufficed, this is very shady business conduct and you will do well to fix it soon.
@S74nk0 , please reopen the ticket.

**Comment by Singman33** [comment]
WTF, are you totally blind or what ?
If you think nicehash is earning lots of money with benchmarking, please use your brain and go to the BTC address statistics. You will suddenly realize why your affirmation is totally useless. That thread is closed, definitively.
This address (34HKWdzLxWBduUfJE9JxaFhoXnfC6gmePG) got 44 BTC since 2015-05-05, it's 0.06 BTC per day because some stupid users never read the documentation and dont change the destination address.
As explained, benchmarks should be done and if you are not happy with that, get the source code and replace with yours ! A program will never be idiot-proof.

**Comment by Scorpers** [comment]
40,000USD since 2015 made on USERS computers without their knowledge since the benchmark windows are hidden, MAKE A CHANGE or give proper warning! 
or explain why you think FRAUD is an acceptable way to make money.

**Comment by S74nk0** [comment]
@Scorpers NHM is Open Source there is nothing hidden. 
![demonotice](https://cloud.githubusercontent.com/assets/1517233/22395729/7928375e-e546-11e6-9142-1f01425c5f86.PNG)

There are people **mining** with this address for a long period of time (they probably didn't read the **DISCLAIMER** or who knows), at the time of writing there are Algorithms **mining** on this address that do not use a direct connection for bench-marking like **DaggerHashimoto** and **Equihash** (offline benchmark). The **DEMO** user wasn't used as the **default** for bench-marking from 2015 ( [commit](https://github.com/nicehash/NiceHashMiner/blob/94436a7fa2f61ebcde54fef561d42002aacdcf36/NiceHashMiner/Miners/sgminer.cs#L148) you can see demo user was a fallback, this is 22 Sep 2016). With major rewrites to the code and to reduce global state and all the hidden BUGs that can come with it this was removed completely (in one of the future commits from 22 Sep 2016). And there aren't any significant number of shares (**if any**) to be made when bench-marking (those numbers come from people mining in the **DEMO** mode and they got the **notice**). They clicked yes for different reasons: **wanted to see the demo**, click yes reflex, don't like to read, can't read, doesn't understand English/Russian, ... so they have been noted but how did or if they did get the information... a program cannot handle this human behavior.

> 40,000USD since 2015

Your estimate is way off. You are taking into account the current BTC to USD exchange and you aren't accounting BTC volatility, the day of payouts... 

> the benchmark windows are hidden

This is a technical thing. We cannot pipe the output and parse the data if they are visible.

> MAKE A CHANGE or give proper warning!

We'll update the Terms Of Use

> explain why you think FRAUD is an acceptable way to make money

Most software NH controls is open source, stats are transparent, regular payouts, ... things are transparent.

I hope this answers your question and reduces doubts.

**Comment by Scorpers** [comment]
although a simple pop-up would have sufficed in case no BTC address is given - great that there is one but its only a small part of the larger issue, this is not changed if there is an address given by the user, so even if u do enter an address it uses the "demo" address, that`s the part i find severely misleading that needs addressing.

**Comment by Eyedol-X** [comment]
I wondered about this for a while now but never looked into it. On my multi-GPU Rigs it takes a while to do the benchmarks completely and IMO it should use the BTC address that I have put into NH Miner if one exists at the time of the benchmark.

If the benchmark cannot be run locally and requires connecting to a server to submit actual shares, it should go to the address I provided. If there isn't an address for new users as described above then it should default to the "Demo" address.

I'm sure NH Miner can be updated to reflect this type of change but the question is, will they make this update?

Another Scenario: If I'm testing OC's or Bios mods and I want to see how it performs on the NH Benchmark, that could be a lot of free benchmarking shares that NH is getting for me simply trying to see if it performs any better with their miner.

TLDR: IMO NH needs to update their miner to use the BTC address provided by the miner if one exists at the time of the benchmark. If no BTC address exists at the time of the benchmark, it should use the demo address with no user alert or interaction required.

Lastly, I don't believe this is a scenario of Fraud. I think this is just an opportunity to either:
A. Better communicate to miners that benchmarking shares are forfeit (not sure how well this will be received once the mass of users learn this, could go either way)
B. Update the miner with changes that benefit the miner and NH. Right now NH is the only benefactor of shares submitted via benchmarking.

**Comment by bathrobehero** [comment]
Copy pasting from btctalk:
I dislike nicehash a lot but come on guys, at least check shit out before accusing them.

Here are all the incoming transactions to that address (reverse chronological order):

0.00020728 BTC
0.00020707 BTC
0.00041969 BTC
0.00021641 BTC
0.00021065 BTC
0.00021423 BTC
0.00021226 BTC
0.00020601 BTC
0.00020922 BTC
0.00021408 BTC
0.0002092 BTC
0.00020454 BTC
0.00016164 BTC
0.00018588 BTC
0.00017548 BTC
_3.94271301 BTC
14.30226584 BTC
8 BTC
8 BTC
0.28751105 BTC
9.71248895 BTC
0.1 BTC_


The address was inactive for almost 17 months after the 3.94 BTC transaction so they made exactly 0.003254 BTC since then from all the benchmarks with 0.00157894 BTC unpaid balance so they _ripped you guys off_ (/s) with a total of
**0.00483294 BTC**.

**Comment by pkumnick** [comment]
Nicehash devs this must be rectified immediately. Not "nice" at all.

This is pure greed even though it is a small amount it is still theft and countless small thefts from many can add up to quite a bit over time.  If it wasn't theft you wouldn't try to hide it and deceptively charge it over and above the fee you say you charge for using the nicehash pools.

I would never have found out about it except I finally got so fed up with your program not working I turned on the debug mode and started paying attention to the debug readouts and realised IT DOESN'T USE MY WALLET ADDRESS!

What's worse about it is that it is a fee that pays for your incompetence. I've been stolen from every time your software screws up and does not complete benchmarking properly, as it seems to fail a lot. and every time I try a new version that doesn't work etc.  there is no technical reason why it can't use the user's wallet address.

**Comment by Eyedol-X** [comment]
> https://github.com/nicehash/NiceHashMiner/issues/556#issuecomment-323054210

I believe in the new version of Nicehash miner legacy they have resolved this issue, if you set your BTC address first, then perform a benchmark, I believe I caught it showing up in my history during the testing. I haven't fully confirmed it yet but I've noticed what I believe was a rig appearing during benchmarking on a couple of different rigs. I can't say I've confirmed this yet because I didn't sit and watch a multi-gpu rig benchmark for an hour but I think some algos, maybe not all are using the supplied address now during benchmarking.
